### PR TITLE
refactor(stdlib): rename percentile to quantile and implement percent…

### LINKED ISSUE #1072: Rename percentile and quantile
Percentile and quantile are basically the same thing except that percentile carries the context of it being a percentage i.e  a number between 0 - 100. We should rename the current percentile function to quantile and add a percentile function that is written in terms of quantile.

```
percentile = (p,tables=<-) => tables |> quantile(q: p / 100)
```

### PR DESCRIPTION
**Pull request by jlapacik**
…ile in terms of quantile

BREAKING CHANGE: Rename the percentile function to quantile. Also rename the percentile
parameter to q. Also implement a new percentile function that takes a parameter p that
is an integer between 0 and 100 inclusive. The new percentile function is defined in
terms of quantile.

Fixes #1072.

### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: docs/SPEC.md [hunk 1]
```diff
@@ -1592,10 +1592,10 @@ Median is defined as:
 
     median = (method="estimate_tdigest", compression=0.0, tables=<-) =>
     	tables
-    		|> percentile(percentile:0.5, method:method, compression:compression)
+    		|> quantile(quantile:0.5, method:method, compression:compression)
```
Comment (nathanielc) [inline docs/SPEC.md]: Below the argument is defined as `q` not `quantile.`

**Comment by jlapacik** [comment]
@nathanielc I think it's probably unnecessary, even though I put up the PR :) . They're exactly the same function, their arguments are just at different scales. I think I prefer to have just one way to perform a quantile calculation. I don't see it as a huge mental burden to switch between a [0, 1] scale vs [0, 100].

**Comment by nathanielc** [comment]
OK, in that case should we just rename it to `quantile` and not have a `percentile` function? There was some confusion around calling it a percentile while using a value between `0-100`.

**Comment by jlapacik** [comment]
I would prefer it called quantile, especially if we are using values between 0 and 1. Currently it's a little inconsistent. But we don't need both functions.

**Pull request merged**